src/test/resources/cliointegrationtest/clio_test.py

```diff
@@ -62,10 +71,86 @@ def test_bad_path():
     assert js['rejection'] == 'The requested resource could not be found.'
 
 
+def __get_field_names_by_type(mapping_properties):
+    field_names_by_type = defaultdict(list)
+    for (field_name, field_properties) in mapping_properties.items():
+        field_type = field_properties['type']
+        field_names_by_type[field_type].append(field_name)
+    for (field_type, field_names) in field_names_by_type.items():
+        field_names.sort()
+    return dict(field_names_by_type)
+
+
+def __assert_expected_fields(actual_fields, expected_fields):
+    assert actual_fields.keys() == expected_fields.keys()
+    for (field_type, actual_field_names) in actual_fields.items():
+        expected_field_names = expected_fields[field_type]
+        assert actual_field_names == expected_field_names
+
+
+def test_readgroups_mapping():
+    r = requests.get(elasticsearch_http_uri + '/readgroups/_mapping/default')
+    js = r.json()
+    mappings = js['readgroups']['mappings']['default']
+    assert mappings['dynamic'] == 'false'
+    mapping_properties = mappings['properties']
+    field_names_by_type = __get_field_names_by_type(mapping_properties)
+    __assert_expected_fields(
+        field_names_by_type,
+        {
```

Comment:
Thanks, I find this much easier to read. We can keep the basic structure if we decide to migrate to scala.